learning_resources/serializers.py

```diff
@@ -438,3 +481,98 @@ class Meta:
             "resource_readable_num",
             "resource_type",
         ]
+
+
+class UserListSerializer(serializers.ModelSerializer, WriteableTopicsMixin):
+    """
+    Simplified serializer for UserList model.
+    """
+
+    item_count = serializers.SerializerMethodField()
+    image = serializers.SerializerMethodField()
+
+    def get_image(self, instance) -> dict:
+        """Return the image of the first item"""
+        list_item = instance.children.order_by("position").first()
+        if list_item and list_item.child.image:
+            return LearningResourceImageSerializer(instance=list_item.child.image).data
+        return None
+
+    def get_item_count(self, instance) -> int:
+        """Return the number of items in the list"""
+        return getattr(instance, "item_count", None) or instance.resources.count()
+
+    def create(self, validated_data):
+        """Ensure that the list is created by the requesting user"""
+        request = self.context.get("request")
+        if request and hasattr(request, "user") and isinstance(request.user, User):
+            validated_data["author"] = request.user
+            topics_data = validated_data.pop("topics", [])
+            with transaction.atomic():
+                userlist = super().create(validated_data)
+                userlist.topics.set(
+                    models.LearningResourceTopic.objects.filter(id__in=topics_data)
+                )
+            return userlist
+        return None
+
+    def update(self, instance, validated_data):
+        """Ensure that the list is authored by the requesting user before modifying"""
+        request = self.context.get("request")
+        validated_data["author"] = request.user
+        topics_data = validated_data.pop("topics", None)
+        with transaction.atomic():
+            userlist = super().update(instance, validated_data)
+            if topics_data is not None:
+                userlist.topics.set(
+                    models.LearningResourceTopic.objects.filter(id__in=topics_data)
+                )
+            return userlist
+
+    class Meta:
+        model = models.UserList
+        exclude = COMMON_IGNORED_FIELDS
+        read_only_fields = ["author", "resources"]
```

Comment:
**Comment:** One thing I noticed is that the `resources` property and the `/userlist/<id>/resources` endpoint don't necessarily return the same order. 

Maybe we could order it or exclude it.